### PR TITLE
Update testing to new approach

### PR DESCRIPTION
**Pull request by ingwarr**
Fixes https://github.com/forseti-security/terraform-google-forseti/issues/237

### CODE REVIEW

File: test/setup/make_source.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+project_id=$(terraform output project_id)
+echo "export TF_VAR_project_id='$project_id'" >> ../source.sh
+echo "export TF_VAR_pubsub_project_id='$project_id'" >> ../source.sh
+
+network_project=$(terraform output network_project)
+echo "export TF_VAR_network_project='$network_project'" >> ../source.sh
+
+enforcer_project_id=$(terraform output enforcer_project_id)
+echo "export TF_VAR_enforcer_project_id='$enforcer_project_id'" >> ../source.sh
+echo "export TF_VAR_sink_project_id='$enforcer_project_id'" >> ../source.sh
```
Comment (kopachevsky) [inline test/setup/make_source.sh]: several echo to same file was causing lint errors for me on other project, did linter ran well here?

Comment (ingwarr) [reply test/setup/make_source.sh]: Yes

File: modules/on_gke/README.md [hunk 2]
```diff
@@ -2,63 +2,28 @@
...
-
-1. Terraform and kubectl are [installed](#software-dependencies) on the machine where Terraform is executed.
-2. The Service Account you execute the module with has the right [permissions](#configure-a-service-account).
-3. The Compute Engine, Kubernetes Engine, and Container Registry APIs are [active](#enable-apis) on the project you will launch the cluster in.
-4. If you are using a Shared VPC, the APIs must also be activated on the Shared VPC host project and your service account needs the proper permissions there.
-5. Forseti has been deployed into a GCP project, preferably the same project in which this Forseti on GKE is intended to be installed.
-
-The [project factory](https://github.com/terraform-google-modules/terraform-google-project-factory) can be used to provision projects with the correct APIs active and the necessary Shared VPC connections.
-
-### Software Dependencies
```
Comment (kopachevsky) [inline modules/on_gke/README.md]: I don't sure regarding this one as I understand we need to move to CONTIBUTION.md all related to develop /extend modules, and all needed to run module - should stay in readme

File: test/setup/iam.tf [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+    "roles/iam.serviceAccountUser",
+    "roles/serviceusage.serviceUsageAdmin",
+    "roles/storage.admin",
+    "roles/cloudsql.admin",
+    "roles/pubsub.admin",
+    "roles/bigquery.dataViewer",
+  ]
+
+  forseti_project_required_roles = [
+    "roles/owner",
```
Comment (kopachevsky) [inline test/setup/iam.tf]: why do we need "roles/owner" as long it include all stuff like compute admin etc?

File: test/setup/iam.tf [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    "roles/compute.admin",
+    "roles/compute.instanceAdmin",
```
Comment (kopachevsky) [inline test/setup/iam.tf]: isn't "roles/compute.admin" already cover compute.*? why do we need "roles/compute.instanceAdmin"

File: test/setup/iam.tf (hunk 3 above)
Comment (ingwarr) [reply test/setup/iam.tf]: I need it for role administration

File: test/setup/iam.tf (hunk 4 above)
Comment (ingwarr) [reply test/setup/iam.tf]: Yes you are right, I'll fix it

File: modules/on_gke/README.md (hunk 2 above)
Comment (ingwarr) [reply modules/on_gke/README.md]: Maybe I'm wrong, but only the central README file is specified in the document...

File: test/setup/iam.tf [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+    "roles/storage.admin", // Permissions to create GCS buckets that the enforcer will manage
+    "roles/billing.projectManager",
+    "roles/pubsub.admin",
+    "roles/compute.admin",
+    "roles/bigquery.dataViewer",
+  ]
+}
+
+resource "random_string" "main" {
+  upper   = "true"
```
Comment (paulpalamarchuk) [inline test/setup/iam.tf]: Variables type is boolean

File: test/setup/iam.tf [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  suffix = random_string.main.result
```
Comment (paulpalamarchuk) [inline test/setup/iam.tf]: What is that variable for?

File: test/setup/pre_prepare.sh [hunk 7]
```diff
@@ -12,8 +14,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ruby '2.5.3'
-
-source 'https://rubygems.org/' do
-  gem 'kitchen-terraform', '~> 4.9'
-end
+# shellcheck disable=SC2018
+project_suffix="$(LC_ALL=C tr -dc 'a-z' < /dev/urandom | fold -w 4 | head -n 1)"
```
Comment (paulpalamarchuk) [inline test/setup/pre_prepare.sh]: It is better to generate this suffix inside terraform. Or use already created `local.suffix`

**Comment by ingwarr** [comment]
@paulpalamarchuk I've solved your comments

File: test/setup/iam.tf (hunk 5 above)
Comment (ingwarr) [reply test/setup/iam.tf]: Removed

File: examples/on_gke_new_gke_cluster/README.md [hunk 8]
```diff
@@ -38,34 +38,4 @@ In order to operate with the Service Account you must activate the following API
...
-| k8s\_forseti\_server\_image\_tag | The tag for the container image for the Forseti server | string | `"latest"` | no |
-| k8s\_tiller\_sa\_name | The Kubernetes Service Account used by Tiller | string | `"tiller"` | no |
-| network\_name | The name of the VPC being created | string | `"gke-network"` | no |
-| production | Whether or not to deploy Forseti on GKE in a production configuration | string | `"true"` | no |
-| project\_id | The ID of an existing Google project where Forseti will be installed | string | n/a | yes |
-| region | Region where forseti subnetwork will be deployed | string | n/a | yes |
-| sub\_network\_name | The names of the subnet being created | string | `"gke-sub-network"` | no |
-| zones | The zones to host the cluster in (optional if regional cluster / required if zonal) | list | `<list>` | no |
-
 <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply examples/on_gke_new_gke_cluster/README.md]: Fixed

File: examples/on_gke_end_to_end/README.md [hunk 9]
```diff
@@ -47,32 +47,3 @@ In order to operate with the Service Account you must activate the following API
...
-| k8s\_tiller\_sa\_name | The Kubernetes Service Account used by Tiller | string | `"tiller"` | no |
-| network\_name | The name of the VPC being created | string | `"gke-network"` | no |
-| org\_id | GCP Organization ID that Forseti will have purview over | string | `""` | no |
-| production | Whether or not to deploy Forseti on GKE in a production configuration | string | `"true"` | no |
-| project\_id | The ID of an existing Google project where Forseti will be installed | string | n/a | yes |
-| region | Region where forseti subnetwork will be deployed | string | n/a | yes |
-| sub\_network\_name | The names of the subnet being created | string | `"gke-sub-network"` | no |
-| zones | The zones to host the cluster in (optional if regional cluster / required if zonal) | list | `<list>` | no |
-
-[^]: (autogen_docs_end)
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply examples/on_gke_end_to_end/README.md]: fixed

[the remark this answers is not in the document]
Comment (ingwarr) [reply examples/on_gke_end_to_end/README.md]: fixed

File: examples/on_gke_new_gke_cluster/README.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ingwarr) [reply examples/on_gke_new_gke_cluster/README.md]: fixed

File: test/setup/iam.tf (hunk 5 above)
Comment (ingwarr) [reply test/setup/iam.tf]: removed

**Comment by ingwarr** [comment]
Tests are broken, sometimes converge passes, but not always because, we have an unstable download speed of forseti repository, `kitchen verify` never passes. I continue an investigation of root causes failing of the tests and trying to fix them. I've fixed SA validation:

Profile: simple-example
Version: (not specified)
Target:  local://

  ✔  forseti-org-iam: Validate organization roles of SA
     ✔  Command: `gcloud organizations get-iam-policy 684124036889 --filter='bindings.members:forseti-server-gcp-581438e6@ci-forseti-serv-nbwy.iam.gserviceaccount.com' --flatten='bindings[].members' --format='json(bindings.role)'` has all expected org roles
     ✔  Command: `gcloud organizations get-iam-policy 684124036889 --filter='bindings.members:forseti-server-gcp-581438e6@ci-forseti-serv-nbwy.iam.gserviceaccount.com' --flatten='bindings[].members' --format='json(bindings.role)'` exit_status should eq 0
     ✔  Command: `gcloud organizations get-iam-policy 684124036889 --filter='bindings.members:forseti-server-gcp-581438e6@ci-forseti-serv-nbwy.iam.gserviceaccount.com' --flatten='bindings[].members' --format='json(bindings.role)'` stderr should eq ""

**Comment by morgante** [comment]
It looks like even converge is failing. Also you need to rebase on master.

**Comment by ingwarr** [comment]
> It looks like even converge is failing. Also you need to rebase on master.
@morgante I'll rebase it. As I remember, we discussed this situation(a periodic failing of converge) last Friday, this is due to too long time of receiving data from the forseti repository to the VM.

File: test/fixtures/bastion/main.tf [hunk 10]
```diff
@@ -55,6 +55,7 @@ resource "google_compute_instance" "main" {
     subnetwork_project = var.project_id
 
     access_config {
+      network_tier = "STANDARD"
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply test/fixtures/bastion/main.tf]: Ok, I'll remove it

**Comment by aaron-lane** [comment]
@ingwarr a patch has been merged to master which should solve the excessive amount of time to clone the Forseti repository. Please rebase against master.

File: test/setup/variables.tf [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+
+variable "folder_id" {
+  description = "The folder to deploy in"
+}
+
+variable "billing_account" {
+  description = "The billing account id associated with the project, e.g. XXXXXX-YYYYYY-ZZZZZZ"
+}
+
+variable "project_suffix" {
```
[the remark this answers is not in the document]
Comment (ingwarr) [reply test/setup/variables.tf]: > @ingwarr a patch has been merged to master which should solve the excessive amount of time to clone the Forseti repository. Please rebase against master.

@aaron-lane Ok, thank you!!! I'll do it!

**Comment by ingwarr** [comment]
@aaron-lane Thank you, I'll rebase it

File: Makefile [hunk 12]
```diff
@@ -12,184 +12,74 @@
...
 		-e SERVICE_ACCOUNT_JSON \
-		-e NETWORK \
-		-e NETWORK_PROJECT \
-		-e SUBNETWORK \
-		-e REGION \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && exec /bin/bash"
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: We should quote `$(CURDIR)`, e.g. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 13]
```diff
@@ -12,184 +12,74 @@
...
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen create"
-
-.PHONY: docker_converge
-docker_converge:
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: We should quote `$(CURDIR)`, e.g. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 14]
```diff
@@ -12,184 +12,74 @@
...
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen converge"
-
-.PHONY: docker_verify
-docker_verify:
+		-e TF_VAR_org_id \
+		-e TF_VAR_folder_id \
+		-e TF_VAR_billing_account \
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: We should quote `$(CURDIR)`, e.g. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 15]
```diff
@@ -12,184 +12,74 @@
...
 		-e SERVICE_ACCOUNT_JSON \
-		-e NETWORK \
-		-e NETWORK_PROJECT \
-		-e SUBNETWORK \
-		-e REGION \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen verify"
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: We should quote `$(CURDIR)`, e.g. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 16]
```diff
@@ -12,184 +12,74 @@
...
-		-e SERVICE_ACCOUNT_JSON \
-		-e NETWORK \
-		-e NETWORK_PROJECT \
-		-e SUBNETWORK \
-		-e REGION \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		/bin/bash -c "source test/ci_integration.sh && setup_environment && kitchen destroy"
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: We should quote `$(CURDIR)`, e.g. `-v "$(CURDIR)":/workspace`

File: Makefile [hunk 17]
```diff
@@ -12,184 +12,74 @@
...
-		-e SERVICE_ACCOUNT_JSON \
-		-e NETWORK \
-		-e NETWORK_PROJECT \
-		-e SUBNETWORK \
-		-e REGION \
-		-e GOOGLE_APPLICATION_CREDENTIALS=${CREDENTIALS_PATH} \
-		-v $(CURDIR):/cft/workdir \
-		${DOCKER_REPO_BASE_KITCHEN_TERRAFORM} \
-		make test_integration
+		-v $(CURDIR):/workspace \
```
Comment (Jberlinsky) [inline Makefile]: We should quote `$(CURDIR)`, e.g. `-v "$(CURDIR)":/workspace`

File: README.md [hunk 18]
```diff
@@ -195,7 +195,7 @@ your environment.
 | sqladmin\_max\_calls | Maximum calls that can be made to SQL Admin API | string | `"1"` | no |
 | sqladmin\_period | The period of max calls for the SQL Admin API (in seconds) | string | `"1.1"` | no |
 | storage\_bucket\_location | GCS storage bucket location | string | `"us-central1"` | no |
-| storage\_disable\_polling | Whetservicemanagement\_disable\_pollingher to disable polling for Storage API | bool | `"false"` | no |
+| storage\_disable\_polling | Whetservicemanagement_disable_pollingher to disable polling for Storage API | bool | `"false"` | no |
```
Comment (Jberlinsky) [inline README.md]: This doesn't look right -- what's going on here?

File: test/integration/simple_example/controls/forseti.rb [hunk 19]
```diff
@@ -14,6 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+# encoding: utf-8
```
Comment (Jberlinsky) [inline test/integration/simple_example/controls/forseti.rb]: Is this line necessary?

File: CONTRIBUTING.md [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+## Generating Documentation for Inputs and Outputs
+
+The Inputs and Outputs tables in the READMEs of the root module,
+submodules, and example modules are automatically generated based on
+the `variables` and `outputs` of the respective modules. These tables
+must be refreshed if the module interfaces are changed.
+
+### Execution
+
+Run `make generate_docs` to generate new Inputs and Outputs tables.
```
[the remark this answers is not in the document]
Comment (Jberlinsky) [reply CONTRIBUTING.md]: Sorry -- missed the alias in `Makefile`. Disregard.

**Pull request merged**